packages/elements/src/tree/elements/tree-item.ts

```diff
@@ -156,24 +157,61 @@ export class TreeItem<T extends TreeDataItem = TreeDataItem> extends ControlElem
     return this.checkedState === CheckedState.INDETERMINATE;
   }
 
+  /**
+   * Handles aria-checked and aria-selected on mode changes
+   * aria-checked is used for multiple mode due to tri-state support
+   * @returns {void}
+   **/
+  private multipleChanged (): void {
+    this.removeAttribute(this.multiple ? 'aria-selected' : 'aria-checked');
+    this.checkedChanged();
+  }
+
+  /**
+   * Handles selected and aria attribute changes
+   * @returns {void}
+   */
+  private checkedChanged (): void {
+    switch (this.checkedState) {
+      case CheckedState.CHECKED:
+        this.setAttribute('selected', '');
+        this.setAttribute(this.multiple ? 'aria-checked' : 'aria-selected', 'true');
+        break;
+      case CheckedState.INDETERMINATE:
+        this.setAttribute('selected', 'indeterminate');
+        this.setAttribute('aria-checked', 'mixed');
+        break;
+      default:
+        this.removeAttribute('selected');
+        this.setAttribute(this.multiple ? 'aria-checked' : 'aria-selected', 'false');
```

Comment:
I think you could move the logic from the render to here.
```suggestion
        if (this.parent && !this.multiple) {
          this.removeAttribute('aria-selected');
        }
        else {
          this.setAttribute(this.multiple ? 'aria-checked' : 'aria-selected', 'false');
        }
```